tf/chunkparser.py

```diff
@@ -218,14 +325,17 @@ def convert_v5_to_tuple(self, content):
             float32 best_m (4 bytes)
             float32 plies_left (4 bytes)
         """
-        (ver, input_format, probs, planes, us_ooo, us_oo, them_ooo, them_oo,
-         stm, rule50_count, dep_ply_count, winner, root_q, best_q, root_d,
-         best_d, root_m, best_m, plies_left) = self.v5_struct.unpack(content)
-        # v3/4 data sometimes has a useful value in dep_ply_count, so copy that over if the new ply_count is not populated.
-        if plies_left == 0:
-            plies_left = dep_ply_count
-        plies_left = struct.pack('f', plies_left)
 
+        # jhorthos query - this is the same as V5 code - still valid?
+        else:
+            (ver, input_format, probs, planes, us_ooo, us_oo, them_ooo, them_oo,
+             stm, rule50_count, dep_ply_count, winner, root_q, best_q, root_d,
+             best_d, root_m, best_m, plies_left) = self.v5_struct.unpack(content)
+            # v3/4 data sometimes has a useful value in dep_ply_count, so copy that over if the new ply_count is not populated.
```

Comment:
If you follow my suggestion regarding always unpacking with v6 struct - keep this code to do with plies_left.  It'll never fire for proper v5 or v6 data, only for v3/4.

Reply:
note that dep_ply_count is called invariance_info in your v6 unpack. So you'll need to rename.

Reply:
fixed, unconditional V6 unpack now